nautobot/project-static/js/forms.js

```diff
@@ -652,6 +655,30 @@ function initializeDynamicFilterForm(context){
         e.preventDefault()
         let dynamic_form = $("#dynamic-filter-form");
         dynamic_form.find(`input[name*="form-"], select[name*="form-"]`).removeAttr("name")
+        const urlParams = new URLSearchParams(window.location.search);
+        const saved_view_pk = urlParams.get('saved_view');
+        var saved_view_pk_query_string = ""
+        if (saved_view_pk != null){
+            saved_view_pk_query_string = "&saved_view=" + saved_view_pk
+        }
+        const sort_order = urlParams.getAll('sort');
+        var sort_order_query_string = ""
+        if (sort_order != null){
+            for (var i = 0; i < sort_order.length; i++){
+                appendage = "&sort=" + sort_order[i]
+                sort_order_query_string += appendage
+            }
+        }
+        const per_page = urlParams.get('per_page');
+        var per_page_query_string = ""
+        if (per_page != null) {
+            per_page_query_string = "&per_page=" + per_page
+        }
+        const table_changes = urlParams.get('table_changes_pending');
+        var table_changes_query_string = ""
+        if (table_changes != null) {
+            table_changes_query_string = "&table_changes_pending=" + table_changes
+        }
```

Comment:
This still feels sketchy to me. Can't we just assign these parameters directly into `search_query` rather than manually constructing a bunch of distinct query strings?

Reply:
Refactored